lib/module.js

```diff
@@ -402,6 +427,34 @@ Module._resolveLookupPaths = function(request, parent) {
   return [id, [path.dirname(parent.filename)]];
 };
 
+// when preserving symlinks, if the main file is both a file symlink AND
+// has execute permission (on windows, its extension is in PATHEXT), then
+// its target is simply read and resolved, potentially relative to the
+// symlink's dirname. in all other cases the original path is preserved.
+Module._resolveMainRequest = function(request) {
+  if(!preserveSymlinks) return request
+
+  request = Module._resolveFilename(request, null)
+  if(!fs.lstatSync(request).isSymbolicLink()) return request
+
+  let isExecutable = false
+  if(process.platform === 'win32') {
+    const execExts = (process.env.PATHEXT || '').split(path.delimiter)
+    const requestExt = path.extname(request).toLowerCase()
+    isExecutable =
+      execExts.some(execExt => requestExt === execExt.trim().toLowerCase())
+  } else try {
```

Comment:
This is confusing, please just include the try-catch inside a normal `else {}` block.